tests/utils_tests/test_module_loading.py

```diff
@@ -188,28 +188,17 @@ class TestFinder:
     def __init__(self, *args, **kwargs):
         self.importer = zipimporter(*args, **kwargs)
 
-    def find_module(self, path):
-        importer = self.importer.find_module(path)
-        if importer is None:
+    def find_spec(self, path, target=None):
+        spec = self.importer.find_spec(path, target)
+        if spec is None:
             return
-        return TestLoader(importer)
-
-
-class TestLoader:
-    def __init__(self, importer):
-        self.importer = importer
-
-    def load_module(self, name):
-        mod = self.importer.load_module(name)
-        mod.__loader__ = self
-        return mod
+        return spec
 
 
 class CustomLoader(EggLoader):
     """The Custom Loader test is exactly the same as the EggLoader, but
-    it uses a custom defined Loader and Finder that is intentionally
-    split into two classes. Although the EggLoader combines both functions
-    into one class, this isn't required.
+    it uses a custom defined Loader class. Although the EggLoader combines both
```

Comment:
An aside: I see the reference to `EggLoader`. Haven't Python "eggs" been pretty much obsolete for ages? Can we remove this or the bits that are related to "eggs"? (Something to create a separate ticket for, if so.)

Reply:
Yes, it's a separate thing.